setup-vault.sh

```diff
@@ -87,14 +87,19 @@ set -o pipefail
     export VAULT_TOKEN="$($VAULT write -field=token "$AUTH_PATH" role_id=@"$ROLE_ID_FILE" secret_id=@"$SECRET_ID_FILE")"
     unset AUTH_PATH
     unset SECRET_ID_FILE
-    unset ROLE_ID
     (set -x
      umask 077
      if ! grep -q foundation "$ROLE_ID_FILE" ; then
        $VAULT read -field=key internalservices/macos/tailscale/key tags=tag:ephemeral-mac-ci ephemeral=true > /var/root/tailscale.token
      fi
     )
 
+    export EAPASS_FILE=~root/eapass
+    $VAULT kv put internalservices/macos/detsys/kv/"$(cat ROLE_ID_FILE)"/password password=@"$EAPASS_FILE"
```

Comment:
TODO: make this work for foundation macs as well, or guard this on being a detsys mac